pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java

```diff
@@ -791,25 +796,74 @@ MetricsConfiguration metricsConfiguration() {
   }
 
   private Optional<PermissioningConfiguration> permissioningConfiguration() throws Exception {
-    if (!permissionsAccountsEnabled && !permissionsNodesEnabled) {
+    Optional<LocalPermissioningConfiguration> localPermissioningConfigurationOptional;
+    Optional<SmartContractPermissioningConfiguration>
+        smartContractPermissioningConfigurationOptional;
+
+    if (permissionsConfigFile() != null) {
+      if (!localPermissionsEnabled()) {
+        logger.warn(
+            "Permissions config file set {} but no local permissions enabled",
+            permissionsConfigFile());
+      }
+    }
+
+    if (permissionsNodesContractAddress != null) {
+      if (!contractPermissionsEnabled()) {
+        logger.warn(
+            "Smart contract address set {} but no contract permissions enabled",
+            permissionsNodesContractAddress);
+      }
+    }
+
+    if (!(localPermissionsEnabled() || contractPermissionsEnabled())) {
       if (rpcHttpApis.contains(RpcApis.PERM) || rpcWsApis.contains(RpcApis.PERM)) {
         logger.warn(
             "Permissions are disabled. Cannot enable PERM APIs when not using Permissions.");
       }
       return Optional.empty();
     }
 
-    final LocalPermissioningConfiguration localPermissioningConfiguration =
-        PermissioningConfigurationBuilder.permissioningConfiguration(
-            getPermissionsConfigFile(), permissionsNodesEnabled, permissionsAccountsEnabled);
+    if (localPermissionsEnabled()) {
+      final LocalPermissioningConfiguration localPermissioningConfiguration =
+          PermissioningConfigurationBuilder.permissioningConfiguration(
+              getPermissionsConfigFile(), permissionsNodesEnabled, permissionsAccountsEnabled);
+      localPermissioningConfigurationOptional = Optional.of(localPermissioningConfiguration);
+    } else {
+      localPermissioningConfigurationOptional = Optional.empty();
+    }
+    if (contractPermissionsEnabled()) {
+      if (permissionsNodesContractAddress != null) {
+        final SmartContractPermissioningConfiguration smartContractPermissioningConfiguration =
+            PermissioningConfigurationBuilder.smartContractPermissioningConfiguration(
+                permissionsNodesContractAddress, permissionsNodesContractEnabled);
+        smartContractPermissioningConfigurationOptional =
+            Optional.of(smartContractPermissioningConfiguration);
+      } else {
+        logger.warn("No contract address specified. Cannot enable contract permissioning.");
```

Comment:
Should this be just a warning? If there is no address, there is no point in enabling smart contract permissioning. And it feels like it is a really bad thing if the user doesn't see the warning and thinks that permissioning has been set.

Reply:
agree. added ParameterException in this case.